docs/commands.md

```diff
@@ -1,68 +1,103 @@
 ```sql
-TS.CREATERULE sourceKey aggType bucketSizeSec destKey
+TS.ADD key timestamp value [RETENTION retentionSecs] [LABELS field value..]
 ```
-* sourceKey - key name for source time series
-* aggType - aggregation type one of the following: avg, sum, min, max, range, count, first, last
-* bucketSizeSec - time bucket for aggregated compaction,
-* destKey - key name for destination time series
-
-> DEST_KEY should be of a `timeseries` type, and should be created before TS.CREATERULE is called.
+* timestamp - unix timestamp (in seconds) or `*` for automatic timestamp (using the system clock)
+* value - sample numeric data value (double)
 
-> *Performance Notice*: if a compaction rule exits on a timeseries `TS.ADD` performance might be reduced, the complexity of `TS.ADD` is always O(M) when M is the amount of compactions rules or O(1).
+Optional args:
+   * retentionSecs - max age for samples compared to current time (in seconds).
+      * Default: the global retenionsecs configuration of the database. If not set, this is 0.
+      * When set to 0, the series will not be trimmed at all
+   * labels - set of key-value pairs that represent metadata labels of the key
 
-## TS.DELETERULE - delete a compaction rule
+#### Examples
 ```sql
-TS.DELETERULE sourceKey destKey
+TS.ADD temperature:2:32 1548149180 26 LABELS sensor_id 2 area_id 32 
+TS.ADD temperature:3:11 1548149180 27 RETENTION 3600
+TS.ADD temperature:3:11 1548149181 30
 ```
 
-* sourceKey - key name for source time series
-* destKey - key name for destination time series
+#### Complexity
+If a compaction rule exits on a timeseries `TS.ADD` performance might be reduced, the complexity of `TS.ADD` is always O(M) when M is the amount of compactions rules or O(1).
 
+#### Notes
+You can use this command to add data to an non existing timeseries in a single command.  This is the reason why the labels and retentionsecs are optional arguments.  When specified, RedisTimeSeries will check if it needs to update the labels and or retentionSecs which introduces additional complexity.
 
-## TS.ADD - append a new value to the series
+### TS.INCRBY/TS.DECRBY - Increment the latest value
 ```sql
-TS.ADD key [labels] timestamp value
+TS.INCRBY key value [RESET time-bucket] [RETENTION retentionSecs] [LABELS field value..]
 ```
-* labels - set of key-value pairs that represent metadata labels of the key. This will be used if the module was started
-with default settings. see TODO.
-* timestamp - unix timestamp (in seconds) or `*` for automatic timestamp (using the system clock)
+or
+```sql
+TS.DECRBY key value [RESET time-bucket] [RETENTION retentionSecs] [LABELS field value..]
+```
+This command can be used as a counter/gauge that get automatic history as a time series.
+
+* key - key name for timeseries
 * value - sample numeric data value (double)
 
-### Examples
+Optional args:
+   * time-bucket - time bucket for resetting the current counter in seconds
```

Comment:
resetting

Reply:
Don't get it, do you want to clarify that it goes to 0?